src/main/java/edu/kis/powp/jobs2d/command/gui/CommandManagerWindow.java

```diff
@@ -4,107 +4,150 @@
+        commandFactoryList = new JList(commandFactoryModel);
+        JScrollPane scrollPane = new JScrollPane(commandFactoryList);
+        commandFactoryList.setSelectionMode(ListSelectionModel.SINGLE_INTERVAL_SELECTION);
+        addToWindow(scrollPane);
+        JLabel lastCommandsLabel = new JLabel("Last commands");
+        addToWindow(lastCommandsLabel);
+        Vector<DriverCommand> commands = new Vector<>();
+        lastCommands = new JComboBox(commands);
+        this.commandManager.getChangePublisher().addSubscriber(() -> {
+//            String currentCommand = commandManager.getCurrentCommandString().replace("CompoundCommand name: ", "");
+            DriverCommand currentCommand = commandManager.getCurrentCommand();
+            commands.add(currentCommand);
+
+            lastCommands.setSelectedItem(currentCommand);
+        });
+        addToWindow(lastCommands);
+        btnAddSelectedCommand = new JButton("Add command to factory");
+        btnAddSelectedCommand.addActionListener((ActionEvent e) -> {
+            try {
+                DriverCommand selectedCommand = (DriverCommand)lastCommands.getSelectedItem();
+                assert selectedCommand != null;
+                if(!commandFactoryModel.contains(selectedCommand))
+                    commandFactoryModel.add(commandFactoryModel.getSize(), selectedCommand);
+                commandFactory.addCommand(selectedCommand);
+            } catch (CloneNotSupportedException excp) {
+                excp.printStackTrace();
+            }
+        });
+        addToWindow(btnAddSelectedCommand);
+
+        btnDeleteSelectedCommand = new JButton("Delete command from factory");
+        btnDeleteSelectedCommand.addActionListener((ActionEvent e) ->
+        {
+            commandFactory.removeCommand(commandFactoryList.getSelectedValue().toString());
+            commandFactoryModel.remove(commandFactoryList.getSelectedIndex());
+        });
+        addToWindow(btnDeleteSelectedCommand);
+
+        btnSetSelectedCommand = new JButton("Set selected command");
+        btnSetSelectedCommand.addActionListener((ActionEvent e) ->
+                {
+                    try {
+                        if (commandFactoryList.getSelectedIndex() != -1) {
+                            DriverCommand command = commandFactory.getCommand(commandFactoryList.getSelectedValue().toString());
+                            commandManager.setCurrentCommand(command);
+                        }
+                    } catch (CloneNotSupportedException ex) {
+                        ex.printStackTrace();
+                    }
+                }
+        );
+        addToWindow(btnSetSelectedCommand);
+        btnRunSelectedCommand = new JButton("Run selected command");
+        btnRunSelectedCommand.addActionListener((ActionEvent e) ->
+                {
+                    commandManager.runCurrentCommand();
+                }
+        );
+        addToWindow(btnRunSelectedCommand);
     }
```

Comment:
This method is too large. Should be split into several methods to make it easier to comprehend and manage this factory feature.